client/src/components/project/ProjectForm.vue

```diff
@@ -0,0 +1,157 @@
+            <span class="md-error">{{this.dateErrorMessage}}</span>
+          </md-field>
+
+          <assign-project-employee v-model="employees"/>
+
+          <md-button class="md-raised" type="cancel" @click="$emit('input', false)">CANCEL</md-button>
+          <md-button class="md-raised md-primary" :disabled="!validateForm" type="submit">SUBMIT</md-button>
+        </form>
+      </div>
+
+      <md-snackbar md-position="left" :md-duration="snackbarDuration" :md-active.sync="showSnackbar" md-persistent>
+        <span>{{snackbarMessage}}</span>
+      </md-snackbar>
+  </div>
+</template>
+
+<script>
+import { projectService } from '../../service/project'
+import AssignProjectEmployee from './AssignProjectEmployee.vue'
+export default {
+  name: 'ProjectForm',
+  components: { AssignProjectEmployee },
+  props: ['value', 'editableProject'],
+  data () {
+    return {
+      name: '',
+      description: '',
+      date: '',
+      employees: [],
+      showSnackbar: false,
+      snackbarDuration: 5000,
+      snackbarMessage: '',
+      nameErrorMessage: '',
+      decriptionErrorMessage: '',
+      dateErrorMessage: '',
+      formTitle: 'Add Project'
+    }
+  },
+  computed: {
+    nameMessageClass () {
+      return {
+        'md-invalid': this.name.length < 10 || !validateAlphanumeric(this.name)
+      }
+    },
+    descriptionMessageClass () {
+      return {
+        'md-invalid': this.description.length < 50
+      }
+    },
+    dateMessageClass () {
+      return {
+        'md-invalid': !validateDate(this.date)
+      }
+    },
+    validateForm () {
+      return this.name.length >= 10 && validateAlphanumeric(this.name)
+    }
+  },
+  watch: {
+    name () {
+      if (this.name.length < 10 || !validateAlphanumeric(this.name)) {
+        this.nameErrorMessage = 'The name should be in 10 to 80 letter alphanumeric'
+      }
+    },
+    description () {
+      if (this.description.length < 50) {
+        this.decriptionErrorMessage = 'The description should be in 50 to 300 letter alphabet'
+      }
+    },
+    date () {
+      if (!validateDate(this.date)) {
+        this.dateErrorMessage = 'The date should be in yyyy-mm-dd format'
+      }
+    },
+    editableProject () {
+      if (this.editableProject !== undefined && this.editableProject) {
+        this.name = this.editableProject.name
+        this.description = this.editableProject.description
+        this.date = this.editableProject.date
+        this.employees = this.editableProject.employees
+        this.formTitle = 'Edit Project'
+      }
+    }
+  },
+  methods: {
+    async addProject (values) {
+      const obj = {name: this.name, description: this.description, date: this.date, employees: this.employees}
+      const result = (this.editableProject !== undefined && this.editableProject) ? await projectService.update(this.editableProject.id, obj) : await projectService.add(obj)
```

Comment:
Twice the same logic... `this.editableProject` will return `false` if `undefined` anyway, so no need for the left part of the `&&` condition.

Reply:
done